diff --git a/node_modules/metro-react-native-babel-preset/src/configs/main.js b/node_modules/metro-react-native-babel-preset/src/configs/main.js
index d4d9b34..c61199b 100644
--- a/node_modules/metro-react-native-babel-preset/src/configs/main.js
+++ b/node_modules/metro-react-native-babel-preset/src/configs/main.js
@@ -68,7 +68,7 @@ const defaultPlugins = [
   [require("@babel/plugin-transform-parameters")],
   [require("@babel/plugin-transform-shorthand-properties")],
   [require("@babel/plugin-transform-react-jsx")],
-  [require("@babel/plugin-transform-regenerator")],
+  // [require("@babel/plugin-transform-regenerator")],
   [require("@babel/plugin-transform-sticky-regex")],
   [require("@babel/plugin-transform-unicode-regex")]
 ];
@@ -115,7 +115,7 @@ const babelRuntime = [
   require("@babel/plugin-transform-runtime"),
   {
     helpers: true,
-    regenerator: true
+    regenerator: false
   }
 ];
 
