diff --git a/node_modules/react-native-vector-icons/lib/create-icon-set.js b/node_modules/react-native-vector-icons/lib/create-icon-set.js
index 019313b..5b93bf2 100644
--- a/node_modules/react-native-vector-icons/lib/create-icon-set.js
+++ b/node_modules/react-native-vector-icons/lib/create-icon-set.js
@@ -9,8 +9,8 @@ import {
 } from './react-native';
 
 import createIconButtonComponent from './icon-button';
-import createTabBarItemIOSComponent from './tab-bar-item-ios';
-import createToolbarAndroidComponent from './toolbar-android';
+// import createTabBarItemIOSComponent from './tab-bar-item-ios';
+// import createToolbarAndroidComponent from './toolbar-android';
 
 const NativeIconAPI =
   NativeModules.RNVectorIconsManager || NativeModules.RNVectorIconsModule;
@@ -164,6 +164,7 @@ export default function createIconSet(glyphMap, fontFamily, fontFile) {
   }
 
   Icon.Button = createIconButtonComponent(Icon);
+  /*
   Icon.TabBarItem = createTabBarItemIOSComponent(
     IconNamePropType,
     getImageSource
@@ -173,6 +174,7 @@ export default function createIconSet(glyphMap, fontFamily, fontFile) {
     IconNamePropType,
     getImageSource
   );
+  */
   Icon.getImageSource = getImageSource;
   Icon.loadFont = loadFont;
   Icon.hasIcon = hasIcon;
