diff --git a/node_modules/react-native-vector-icons/lib/create-icon-set.js b/node_modules/react-native-vector-icons/lib/create-icon-set.js
index a0421cc..1fa78ba 100644
--- a/node_modules/react-native-vector-icons/lib/create-icon-set.js
+++ b/node_modules/react-native-vector-icons/lib/create-icon-set.js
@@ -10,8 +10,8 @@ import {
 
 import ensureNativeModuleAvailable from './ensure-native-module-available';
 import createIconButtonComponent from './icon-button';
-import createTabBarItemIOSComponent from './tab-bar-item-ios';
-import createToolbarAndroidComponent from './toolbar-android';
+// import createTabBarItemIOSComponent from './tab-bar-item-ios';
+// import createToolbarAndroidComponent from './toolbar-android';
 
 export const NativeIconAPI =
   NativeModules.RNVectorIconsManager || NativeModules.RNVectorIconsModule;
@@ -167,6 +167,7 @@ export default function createIconSet(
   }
 
   Icon.Button = createIconButtonComponent(Icon);
+  /*
   Icon.TabBarItem = createTabBarItemIOSComponent(
     IconNamePropType,
     getImageSource
@@ -176,6 +177,7 @@ export default function createIconSet(
     IconNamePropType,
     getImageSource
   );
+  */
   Icon.getImageSource = getImageSource;
   Icon.loadFont = loadFont;
   Icon.hasIcon = hasIcon;
