diff --git a/node_modules/react-native-safe-area-context/android/gradle.properties b/node_modules/react-native-safe-area-context/android/gradle.properties
index c9d3fbb..9880579 100755
--- a/node_modules/react-native-safe-area-context/android/gradle.properties
+++ b/node_modules/react-native-safe-area-context/android/gradle.properties
@@ -2,5 +2,5 @@ ReactNativeSafeAreaView_compileSdkVersion=28
 ReactNativeSafeAreaView_buildToolsVersion=28.0.3
 ReactNativeSafeAreaView_targetSdkVersion=27
 ReactNativeSafeAreaView_minSdkVersion=16
-android.useAndroidX=true
-android.enableJetifier=true
+android.useAndroidX=false
+android.enableJetifier=false
diff --git a/node_modules/react-native-safe-area-context/android/src/main/java/com/th3rdwave/safeareacontext/SafeAreaView.java b/node_modules/react-native-safe-area-context/android/src/main/java/com/th3rdwave/safeareacontext/SafeAreaView.java
index 8701a35..ab3c44c 100644
--- a/node_modules/react-native-safe-area-context/android/src/main/java/com/th3rdwave/safeareacontext/SafeAreaView.java
+++ b/node_modules/react-native-safe-area-context/android/src/main/java/com/th3rdwave/safeareacontext/SafeAreaView.java
@@ -13,7 +13,7 @@ import android.view.WindowManager;
 import com.facebook.infer.annotation.Assertions;
 import com.facebook.react.views.view.ReactViewGroup;
 
-import androidx.annotation.Nullable;
+import javax.annotation.Nullable;
 
 public class SafeAreaView extends ReactViewGroup implements ViewTreeObserver.OnGlobalLayoutListener {
   public interface OnInsetsChangeListener {
diff --git a/node_modules/react-native-safe-area-context/android/src/main/java/com/th3rdwave/safeareacontext/SafeAreaViewManager.java b/node_modules/react-native-safe-area-context/android/src/main/java/com/th3rdwave/safeareacontext/SafeAreaViewManager.java
index dcc2a8c..b1f7861 100644
--- a/node_modules/react-native-safe-area-context/android/src/main/java/com/th3rdwave/safeareacontext/SafeAreaViewManager.java
+++ b/node_modules/react-native-safe-area-context/android/src/main/java/com/th3rdwave/safeareacontext/SafeAreaViewManager.java
@@ -8,23 +8,23 @@ import com.facebook.react.uimanager.events.EventDispatcher;
 
 import java.util.Map;
 
-import androidx.annotation.NonNull;
+import javax.annotation.Nonnull;
 
 public class SafeAreaViewManager extends ViewGroupManager<SafeAreaView> {
   @Override
-  @NonNull
+  @Nonnull
   public String getName() {
     return "RNCSafeAreaView";
   }
 
   @Override
-  @NonNull
-  public SafeAreaView createViewInstance(@NonNull ThemedReactContext context) {
+  @Nonnull
+  public SafeAreaView createViewInstance(@Nonnull ThemedReactContext context) {
     return new SafeAreaView(context);
   }
 
   @Override
-  protected void addEventEmitters(@NonNull ThemedReactContext reactContext, @NonNull final SafeAreaView view) {
+  protected void addEventEmitters(@Nonnull ThemedReactContext reactContext, @Nonnull final SafeAreaView view) {
     final EventDispatcher dispatcher =
         reactContext.getNativeModule(UIManagerModule.class).getEventDispatcher();
     view.setOnInsetsChangeListener(new SafeAreaView.OnInsetsChangeListener() {
diff --git a/node_modules/react-native-safe-area-context/ios/.DS_Store b/node_modules/react-native-safe-area-context/ios/.DS_Store
index 1f44500..f0e33d4 100644
Binary files a/node_modules/react-native-safe-area-context/ios/.DS_Store and b/node_modules/react-native-safe-area-context/ios/.DS_Store differ
