diff --git a/node_modules/react-native-gesture-handler/GestureComponents.web.js b/node_modules/react-native-gesture-handler/GestureComponents.web.js
index 9d2a339..539901f 100644
--- a/node_modules/react-native-gesture-handler/GestureComponents.web.js
+++ b/node_modules/react-native-gesture-handler/GestureComponents.web.js
@@ -4,7 +4,7 @@ import {
   FlatList as RNFlatList,
   Switch as RNSwitch,
   TextInput as RNTextInput,
-  ToolbarAndroid as RNToolbarAndroid,
+  // ToolbarAndroid as RNToolbarAndroid,
   ScrollView as RNScrollView,
 } from 'react-native';
 
@@ -20,7 +20,7 @@ export const Switch = createNativeWrapper(RNSwitch, {
   disallowInterruption: true,
 });
 export const TextInput = createNativeWrapper(RNTextInput);
-export const ToolbarAndroid = createNativeWrapper(RNToolbarAndroid);
+export const ToolbarAndroid = null // createNativeWrapper(RNToolbarAndroid);
 export const DrawerLayoutAndroid = createNativeWrapper(RNDrawerLayoutAndroid, {
   disallowInterruption: true,
 });
